controller/podcontroller.go

```diff
@@ -131,38 +132,23 @@ func (c *PodController) onUpdate(oldObj, newObj interface{}) {
 
 	//handle the case when a pg database pod is updated
 	if isPostgresPod(newpod) {
-		// Handle the "role" label change from "replica" to "master" following a failover.  Specifically, patch the pgBackRest
-		// dedicated repository host (specifically the PGBACKREST_DB_PATH env var) to point to the proper directory for the new
-		// primary
-		if oldpod.ObjectMeta.Labels[config.LABEL_PGHA_ROLE] == "promoted" && labels[config.LABEL_PGHA_ROLE] == "master" {
-			err = backrest.UpdateDBPath(c.PodClientset, &pgcluster, labels[config.LABEL_DEPLOYMENT_NAME],
-				newpod.ObjectMeta.Namespace)
-			if err != nil {
-				log.Error(err)
+		// Handle the "role" label change from "replica" to "master" following a failover.
+		// Specifically, take a backup to ensure there is a fresh backup for the cluster
+		// post-failover.
+		if oldpod.ObjectMeta.Labels[config.LABEL_PGHA_ROLE] == "promoted" &&
+			labels[config.LABEL_PGHA_ROLE] == "master" &&
+			pgcluster.Status.State == crv1.PgclusterStateInitialized {
+
+			//look up the backrest-repo pod name
+			selector := config.LABEL_PG_CLUSTER + "=" + clusterName + "," +
+				config.LABEL_PGO_BACKREST_REPO + "=true"
```

Comment:
I find using `fmt.Sprintf` to be a bit cleaner in deciphering what's going on:

```go
selector := fmt.Sprintf("%s=%s,%s=true", config.LABEL_PG_CLUSTER, clusterName, config.LABEL_PGO_BACKREST_REPO)
```

You can quickly determine both what the selector looks like, and which variables you care about.
```